.github/workflows/precommit.yml

```diff
@@ -0,0 +1,21 @@
+# https://pre-commit.com
+# This GitHub Action assumes that the repo contains a valid .pre-commit-config.yaml file.
+# Using pre-commit.ci is even better that using GitHub Actions for pre-commit.
+name: pre-commit
+on:
+  pull_request:
+    branches: [master]
+  push:
+  workflow_dispatch:
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - run: pip install pre-commit
+      - run: pre-commit --version
+      - run: pre-commit install
+      - run: pre-commit run --all-files
```

Comment:
```suggestion
      - uses: actions/setup-python@v5
        with:
          python-version: 3
      - uses: pre-commit/action@v3.0.0
```

Reply:
https://github.com/pre-commit/action is in maintenance-only mode and will not be accepting new features so I tend not to use it.

Reply:
I know it's in maintenance-only mode, but it works – it's also just, well, [20 lines](
https://github.com/pre-commit/action/blob/main/action.yml).

I don't prefer pre-commit.ci because it's an one-man project (and for projects where "auto-fix PRs" is enabled, they tend to mess with my workflow).

Reply:
If auto-fix PRs are messing up your workflow then you do not have pre-commit properly installed locally.

Reply:
@cclauss Well, for a repo like this where every commit should have a `--signoff` to adhere to a CLA, the pre-commit commit wouldn't be appropriate, etc. 😁